backend/game/cards/keep_cards/energy_manipulation_cards/test_energy_hoarder.py

```diff
@@ -0,0 +1,18 @@
+from game.cards.keep_cards.energy_manipulation_cards.energy_hoarder import EnergyHoarder
+
+
+def test_energy_hoarder_player_add_card(player):
+    EnergyHoarder().immediate_effect(player, None)
+    assert player.card_on_hand.get("Energy Hoarder") == \
+           "You gain 1[Star] for every 6[Energy] you have at the end of your turn."
```

Comment:
Asserting on a whole string like this makes for an extremely fragile test.
What's the point of this test? That the player has the card? That's not the assert.

Reply:
the assert confirms that the card was added

Reply:
That's not how the assert READS.
The assert checks if the string is perfectly accurate.
Make the assert simply check if the card was added.

Reply:
Can you explain further, already checking based on the the card dict key value pair for card, key "Energy Hoarder" checking for value  "You gain 1[Star] for every 6[Energy] you have at the end of your turn.", by checking card store card on hand

Reply:
You should not have to assert on a STRING FIELD inside of an OBJECT to assert that the player has the card.

Reply:
string values have been removed

Reply:
gone